### ISSUE TITLE
Upgrade GitLab Pages support to Pelican 4.0.1

### ISSUE THREAD

**Issue by davidag** [getpelican/pelican#2572]
[GitLab has a Pages feature](https://about.gitlab.com/product/pages/), which is similar to GitHub Pages. I've detected a couple of problems using Pelican on GitLab Pages:

- [x] There are [templates for many static site generators](https://gitlab.com/pages), including Pelican, but this one is outdated.
- [ ] When [a new project is created](https://gitlab.com/projects/new), you can _create from template_, but `Pages/Pelican` does not appear. These templates are called [built-in templates](https://gitlab.com/gitlab-org/project-templates/contributing) and are distributed with GitLab.

I've submitted a PR for the first issue: https://gitlab.com/pages/pelican/merge_requests/9

I'm looking into the second one.

**Comment by davidag** [comment]
The merge request was merged, so the first issue is fixed :tada: 

With respect to the second issue, according to the repository owner, [he will submit it to be included as a built-in template in GitLab.](https://gitlab.com/pages/pelican/merge_requests/9#note_174368650)

**Comment by davidag** [comment]
[An issue has been created](https://gitlab.com/gitlab-org/project-templates/contributing/issues/1) in the `project-templates` repo to handle the addition of Pelican to GitLab Pages. Let's hope this keeps the momentum :)

**Comment by justinmayer** [comment]
Thanks for the updates on this endeavor, David. Would you like to update the template(s) from Pelican 4.0.1 to the recently-released version 4.1?

**Comment by davidag** [comment]
[Opened MR with updated template as generated by Pelican 4.1.0 customized for GitLab Pages](https://gitlab.com/pages/pelican/merge_requests/12)

When this is merged, I'll try to push again for Pelican to be included as a built-in template :crossed_fingers:

**Comment by davidag** [comment]
Let's recap:
- [Example Pelican project updated to 4.1.0](https://gitlab.com/pages/pelican) after merging [the corresponding MR](https://gitlab.com/pages/pelican/merge_requests/12)
- [No news on the issue to add Pelican as a project template](https://gitlab.com/gitlab-org/project-templates/contributing/issues/1)

**Comment by justinmayer** [comment]
Many thanks for the update, David. Following is a minor update of my own… 😉 

Thanks to my new project [AutoPub](https://github.com/autopub/autopub), Pelican 4.1.1 was automatically released a few days ago. We try to use semantic versioning, and befitting the patch version designation, the changes in 4.1.1 are minor, so I don't think there's any need to rush out to submit another MR to GitLab. But in case you hadn't heard, I at least wanted to let you know that a new version has been released. And now that Pelican releases are automated, the release cadence should be noticeably more frequent.

To that end… After taking a belated look at your GitLab MR (and I must apologize for reviewing it _after_ it was merged and not before), I noticed two things that might be adjusted to account for the new turbo-charged release cadence:

1. Instead of pinning via `pelican == 4.1.0` in the [requirements file]( https://gitlab.com/pages/pelican/merge_requests/12/diffs#19359a61ae2446b51b549167b014da2fcf265768_2_1), you might consider changing the [version specifier](https://www.python.org/dev/peps/pep-0440/#version-specifiers) to instead utilize the [compatible release](https://www.python.org/dev/peps/pep-0440/#compatible-release) version clause: `pelican ~= 4.1.0`. That should be equivalent to `>= 4.1.0, == 4.1.*` and should obviate the need to submit an additional MR every time AutoPub publishes a new Pelican patch release.

2. Similarly, when [linking to the Pelican docs](https://gitlab.com/pages/pelican/merge_requests/12/diffs#8ec9a00bfd09b3190ac6b22251dbb1aa95a0579d_79_72), you might consider linking to the [stable docs](https://docs.getpelican.com/en/stable/install.html) instead of the [version 4.1.0 docs](https://docs.getpelican.com/en/4.1.0/install.html). That way, the link to the documentation won't need to be updated after every tiny patch release.

What do you think?

**Comment by davidag** [comment]
Thanks for the feedback, Justin!

About the version specifier, my intention when adding it was to avoid incompatibilities between Pelican core and the skeleton files. This leads to the problems that you pointed out.

The thing is that without a criterion of when major, minor and patch version numbers will change, it's difficult to be sure that a new Pelican core won't be incompatible with previously generated skeleton files. Sorry if the criterion exists, but I couldn't find it easily, maybe it's a convention, maybe it's not written yet.

What I try to say is that I would prefer to use `pelican ~= 4.0` making MRs even less necessary :relaxed: but it doesn't seem like a wise move right now.

Beyond the versioning criteria, it seems that changes in the generated skeleton files should be available in the example GitLab repository as quickly as possible. There's no clear way to know when this kind of changes happens. Maybe a special keyword in the changelog could help?

As for the documentation, it would be great to have a version 4.1 (or even 4) link that would redirect to the latest 4.1.x (or 4.x) docs, but I guess that's not easily doable.

    https://docs.getpelican.com/en/4.1/install.html -> https://docs.getpelican.com/en/4.1.1/install.html

Summing up, your suggestions look very reasonable and I'll be happy to push them. Just wanted to let you know how the situation could be improved even more, in my opinon :relaxed: 

PD: Congrats on the AutoPub project, I didn't know it was yours! I can understand why you created it :tada:

**Comment by axilleas** [comment]
:wave: I can help with getting Pelican as a Pages template :) Is there a logo in SVG?

**Comment by davidag** [comment]
@axilleas That's great!! I'm not sure if there's any Pelican logo (though it seems like a nice idea :) Maybe @justinmayer can let us know.

**Comment by davidag** [comment]
@axilleas I think this is the logo: <https://twitter.com/getpelican/photo> but I couldn't find it in Pelican's github repositories.

**Comment by justinmayer** [comment]
@axilleas: Many thanks for reaching out. I'm working on putting up an SVG version of the logo and will post an update here as soon as I can.

**Comment by justinmayer** [comment]
SVG version of logo posted: https://gitlab.com/gitlab-org/project-templates/contributing/issues/1#note_218321551

**Comment by axilleas** [comment]
:wave: I started working on adding Pelican in https://gitlab.com/gitlab-org/gitlab/merge_requests/16934, but I hadn't found the time since to continue... I'm hoping to pick it up again soon.

**Comment by justinmayer** [comment]
Hi @axilleas. Have you been able to make any progress on this endeavor?

**Comment by axilleas** [comment]
> Hi @axilleas. Have you been able to make any progress on this endeavor?

@justinmayer unfortunately not yet :/

**Comment by justinmayer** [comment]
@davidag: I'm going to assign this issue to you in order to prevent it from being marked as stale. As a side note, I'll try to get a new Pelican release out soon so it can hopefully be utilized if and when it's added to GitLab Pages.

**Comment by axilleas** [comment]
@justinmayer thanks!

**Comment by davidag** [comment]
@justinmayer: Perfect, I'll update the GitLab template when the version is released.

**Comment by kakila** [comment]
Hi, thanks for all the wrok. I still do not see the Pelican template in GitLab 13, "create from template" option.

**Comment by axilleas** [comment]
> Hi, thanks for all the work. I still do not see the Pelican template in GitLab 13, "create from template" option.

Yeah, I still haven't got any time to work on it :(

**Comment by justinmayer** [comment]
@axilleas: Now that Pelican 4.5 has been released, I thought I'd follow up regarding this issue. Do you think you might be able to carve out a bit of time to add Pelican to GitLab Pages?

**Comment by mat-m** [comment]
@justinmayer : @axilleas  upgraded to Pelican 4.7 4 months ago: https://gitlab.com/pages/pelican/. 

I guess you can close this issue.

**Comment by axilleas** [comment]
@justinmayer I'm happy to report that I managed to get the Pelican template merged into GitLab 😄 This will be part of GitLab 15.1 released on June 22nd 🙂 See https://gitlab.com/gitlab-org/gitlab/-/merge_requests/16934. You'll be able to create a Pelican site by using the built-in templates https://docs.gitlab.com/ee/user/project/working_with_projects.html#create-a-project-from-a-built-in-template.

**Comment by axilleas** [comment]
@davidag you can close this issue now 🙂

**Comment by justinmayer** [comment]
Many thanks for getting the Pelican template into GitLab, @axilleas! 🎉

**Comment by Ryder17z** [comment]
No. This should not be considered as complete. See #3263
